lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -154,8 +154,9 @@ sub create_from_settings {
         my $dependency_type = $dependency_definition->{dependency_type};
         for my $id (@$ids) {
             if ($dependency_type eq OpenQA::JobDependencies::Constants::DIRECTLY_CHAINED) {
-                my $parent_worker_class = $job_settings->find({job_id => $id, key => 'WORKER_CLASS'});
-                _handle_directly_chained_dep($parent_worker_class, $id, \%settings);
+                my $parent_worker_class = $job_settings->parent_worker_class($id, $settings->{WORKER_CLASS});
+          # assume we want to use the worker class from the parent here (and not the default which is otherwise assumed)
+                $settings->{WORKER_CLASS} = $parent_worker_class;
```

Comment:
```suggestion
                $settings->{WORKER_CLASS} = $parent_worker_class->value;
```
Also, before the code also handled the situation when the job had no WORKER_CLASS at all

Reply:
ah, nevermind, that's done in the other function